publication-rest/src/test/java/no/unit/nva/publication/update/PublishPublicationHandlerTest.java

```diff
@@ -0,0 +1,175 @@
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertEquals(HTTP_BAD_REQUEST, response.getStatusCode());
+    }
+
+    @Test
+    void shouldReturnBadGatewayWhenUnexpectedExceptionOccurs()
+        throws IOException, BadRequestException, NotFoundException {
+        var publication = createPublication();
+        var request = createRequest(publication);
+
+        resourceService = mock(ResourceService.class);
+        when(resourceService.getResourceByIdentifier(publication.getIdentifier())).thenThrow(new RuntimeException());
+        var handler = new PublishPublicationHandler(resourceService);
+        handler.handleRequest(request, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertEquals(HTTP_BAD_GATEWAY, response.getStatusCode());
+    }
+
+    @Test
+    void shouldReturnOkWhenPublishingPublication()
+        throws IOException, BadRequestException {
+        var publication = createPublication();
+        var request = createRequest(publication);
+
+        handler.handleRequest(request, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertEquals(HTTP_OK, response.getStatusCode());
+    }
+
+    private static Map<String, String> publicationIdentifierPathParam(SortableIdentifier publicationIdentifier) {
+        return Map.of(PUBLICATION_IDENTIFIER_PATH_PARAMETER_NAME, publicationIdentifier.toString());
+    }
+
+    private Publication createUnpublishablePublication() throws BadRequestException, NotFoundException {
+        var publication = createPublication();
+        UserInstance userInstance = UserInstance.fromPublication(publication);
+        Resource.fromPublication(publication).publish(resourceService, userInstance);
+        resourceService.unpublishPublication(publication, userInstance);
+        resourceService.deletePublication(publication, userInstance);
+        return publication;
+    }
+
+    private Publication createPublication() throws BadRequestException {
+        Publication publication = randomNonDegreePublication();
+        return Resource.fromPublication(publication)
+                   .persistNew(resourceService, UserInstance.fromPublication(publication));
+    }
+
+    private InputStream createRequestWithRandomUser(SortableIdentifier publicationIdentifier)
+        throws JsonProcessingException {
+        return new HandlerRequestBuilder<Void>(dtoObjectMapper).withPathParameters(
+                publicationIdentifierPathParam(publicationIdentifier))
+                   .withCurrentCustomer(randomUri())
+                   .withUserName(randomString())
+                   .withPersonCristinId(randomUri())
+                   .withTopLevelCristinOrgId(randomUri())
+                   .build();
+    }
+
+    private InputStream createRequest(Publication publication) throws JsonProcessingException {
```

Comment:
The naming here does not indicate that the user has the correct permissions.

Reply:
I might have derived the permissions from the publication (i.e. the top level, etc.)

Reply:
renaming to `createRequestWithUserWithPermissionsToPublishPublication(publication)`